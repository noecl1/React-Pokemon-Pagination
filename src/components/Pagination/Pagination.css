.button-style {
  border: 1px solid grey;
  padding: 15px;
  margin: 5px;
}
