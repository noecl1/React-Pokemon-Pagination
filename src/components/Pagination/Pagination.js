import React from "react";
import "./Pagination.css";

export default function Pagination({ gotoNextPage, gotoPrevPage }) {
  return (
    <div>
      {gotoPrevPage && (
        <button
          data-testid="gotoPreviousButton"
          className="button-style"
          onClick={gotoPrevPage}
        >
          Previous
        </button>
      )}
      {gotoNextPage && (
        <button
          data-testid="gotoNextButton"
          className="button-style"
          onClick={gotoNextPage}
        >
          Next
        </button>
      )}
    </div>
  );
}
