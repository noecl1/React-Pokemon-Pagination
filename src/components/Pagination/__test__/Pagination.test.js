import React from "react";
import ReactDOM from "react-dom";
import Pagination from "../Pagination";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Renders Pagination Button without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Pagination></Pagination>, div);
});

it("Renders Pagination Button correctly", () => {
  const next = "https://pokeapi.co/api/v2/pokemon?offset=40&limit=20";
  const previous = "https://pokeapi.co/api/v2/pokemon?offset=0&limit=20";
  const { getByTestId } = render(
    <Pagination gotoNextPage={next} gotoPrevPage={previous}></Pagination>
  );
  expect(getByTestId("gotoPreviousButton")).toHaveTextContent("Previous");
  expect(getByTestId("gotoNextButton")).toHaveTextContent("Next");
});

it("Pagination Button Matches Snapshot", () => {
  const next = "https://pokeapi.co/api/v2/pokemon?offset=40&limit=20";
  const previous = "https://pokeapi.co/api/v2/pokemon?offset=0&limit=20";
  const tree = renderer
    .create(
      <Pagination gotoNextPage={next} gotoPrevPage={previous}></Pagination>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
