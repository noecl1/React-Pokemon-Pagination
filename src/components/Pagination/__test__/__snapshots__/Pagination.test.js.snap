// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Button Matches Snapshot 1`] = `
<div>
  <button
    className="button-style"
    data-testid="gotoPreviousButton"
    onClick="https://pokeapi.co/api/v2/pokemon?offset=0&limit=20"
  >
    Previous
  </button>
  <button
    className="button-style"
    data-testid="gotoNextButton"
    onClick="https://pokeapi.co/api/v2/pokemon?offset=40&limit=20"
  >
    Next
  </button>
</div>
`;
